dpnp/backend/examples/example_bs.cpp

```diff
@@ -0,0 +1,192 @@
+    dpnp_memory_free_c(d2);
+
+    double* bs_call = (double*)dpnp_memory_alloc_c(size * sizeof(double));
+    dpnp_multiply_c<double, double, double>(S, cdf_d1, bs_call, size);  // S*cdf_d1
+    dpnp_memory_free_c(cdf_d1);
+
+    dpnp_multiply_c<double, double, double>(nrs, T, bs_put, size);            // nrs*T
+    dpnp_exp_c<double, double>(bs_put, bs_put, size);                    // np.exp(nrs*T)
+    dpnp_multiply_c<double, double, double>(K, bs_put, bs_put, size);  // K*np.exp(nrs*T)
+
+    // K*np.exp(nrs*T)*cdf_d2
+    dpnp_multiply_c<double, double, double>(bs_put, cdf_d2, bs_put, size);
+    dpnp_memory_free_c(cdf_d2);
+
+    // S*cdf_d1 - K*np.exp(nrs*T)*cdf_d2
+    dpnp_subtract_c<double, double, double>(bs_call, bs_put, bs_call, size);
+
+    dpnp_multiply_c<double, double, double>(nrs, T, bs_put, size);               // nrs*T
+    dpnp_exp_c<double, double>(bs_put, bs_put, size);                     // np.exp(nrs*T)
+    dpnp_multiply_c<double, double, double>(K, bs_put, bs_put, size);   // K*np.exp(nrs*T)
+    dpnp_subtract_c<double, double, double>(bs_put, S, bs_put, size);   // K*np.exp(nrs*T) - S
+    dpnp_add_c<double, double, double>(bs_put, bs_call, bs_put, size);  // K*np.exp(nrs*T) - S + bs_call
+    dpnp_memory_free_c(bs_call);
+
+    return bs_put;
+}
+
+int main(int, char**)
+{
+    const size_t SIZE = 256;
+
+    const size_t SEED = 7777777;
+    const long SL = 10, SH = 50;
+    const long KL = 10, KH = 50;
+    const long TL = 1,  TH = 2;
+    const double RISK_FREE = 0.1;
+    const double VOLATILITY = 0.2;
+
+    dpnp_queue_initialize_c(QueueOptions::GPU_SELECTOR);
+    std::cout << "SYCL queue is CPU: " << dpnp_queue_is_cpu_c() << std::endl;
+
+    double* S = (double*)dpnp_memory_alloc_c(SIZE * sizeof(double));
+    double* K = (double*)dpnp_memory_alloc_c(SIZE * sizeof(double));
+    double* T = (double*)dpnp_memory_alloc_c(SIZE * sizeof(double));
+    double* result = (double*)dpnp_memory_alloc_c(SIZE * sizeof(double));
+
+    dpnp_rng_srand_c(SEED);
+    dpnp_rng_uniform_c<double>(S, SL, SH, SIZE);  // np.random.uniform(SL, SH, SIZE)
+    dpnp_rng_uniform_c<double>(K, KL, KH, SIZE);  // np.random.uniform(KL, KH, SIZE)
+    dpnp_rng_uniform_c<double>(T, TL, TH, SIZE);  // np.random.uniform(TL, TH, SIZE)
+
+    double* r = (double*)dpnp_memory_alloc_c(1 * sizeof(double));
+    r[0] = RISK_FREE;
+
+    double* sigma = (double*)dpnp_memory_alloc_c(1 * sizeof(double));
+    sigma[0] = VOLATILITY;
+
+    double* rss2 = (double*)dpnp_memory_alloc_c(1 * sizeof(double));
+    rss2[0] = RISK_FREE + VOLATILITY*VOLATILITY/2.;
+
+    double* nr = (double*)dpnp_memory_alloc_c(1 * sizeof(double));
+    nr[0] = -RISK_FREE;
+
+    double* sq2 = (double*)dpnp_memory_alloc_c(1 * sizeof(double));
+    sq2[0] = sqrt(2.);
+
+    double* one = (double*)dpnp_memory_alloc_c(1 * sizeof(double));
+    one[0] = 1.;
+
+    double* two = (double*)dpnp_memory_alloc_c(1 * sizeof(double));
+    two[0] = 2.;
```

Comment:
```suggestion
double* params_for_full = (double*)dpnp_memory_alloc_c(7 * sizeof(double));
params_for_full[0] = RISK_FREE;
params_for_full[1] = VOLATILITY;
params_for_full[2] = RISK_FREE + VOLATILITY*VOLATILITY/2.;
params_for_full[3] = -RISK_FREE;
params_for_full[4] = sqrt(2.);
params_for_full[5] = 1.;
params_for_full[6] = 2.;
```
update on other places and just free `params_for_full` after use.

Please, again, don't do multiple call of mem allocator. This is bad code. Use at once allocated mem and don't call allocator multiple times as much as possible.

Reply:
I agree, but the example won't be comparable with python version with the change.

Reply:
Do you think that python manages mem as you presented? It is disputable, I think.

Reply:
We manage memory the same way in Cython level of the function `full`.

Reply:
Again this is bad case in any way. Python and Cython differs on using mem in some cases be aware. You need to look Cython generation properly for this place of you cython written code.

Also there is no given description about comparison with python version or python-like impl code. Please update the description on src and for PR as well.